Makefile

```diff
@@ -51,6 +51,14 @@ lint:
 	mypy --follow-imports silent --show-error-codes fv3gfs
 	@echo "LINTING SUCCESSFUL"
 
+constraints.txt: requirements.txt setup.py
+	pip-compile $^ --output-file constraints.txt
```

Comment:
I didn't realize this before, but in fv3net we use the `--no-annotate` argument, maybe for similar reasons to why you have some of the sed lines below.

What do these sed lines do? Is the one that removes a `[toml]` extras specification needed?. Can these be removed if we use `--no-annotate`?

Reply:
OK. If there's any way you can avoid the comment at the top from getting one-line-ified I'd appreciate it (I think it's always the same number of lines, so maybe you can skip the first set of lines for the corresponding sed command?), and if you can add comments here about what these lines do that could also help understanding it later, sed is hard.

Reply:
Comment at the top is now expanded and I replace the pip command in the comment by `make constraints.txt`
Verbose the sed commands a bit in Makefile